### ISSUE TITLE
Add the ability to automatically search for materials with the same name on import

### ISSUE THREAD

**Issue by Shatur95** [godotengine/godot-proposals#609]
**Describe the project you are working on:**
[Zoriya](https://github.com/Zoriya), Low Poly 3D RPG

**Describe the problem or limitation you are having in your project:** 
I often need to share materials between several models. For example, Wood material can be applied to a house, tree or weapon. Especially in low poly.

Godot have good support for GLTF, but If I export model without materials, it will have only one material slot in all imported meshes in Godot. If I export model with materials, all materials will be created in the model folder on import in Godot. If I move these materials (for example, move them to a folder with shared materials), the materials will be recreated in the model folder on re-import. Changing material paths is not work, I already submitted [this issue](https://github.com/godotengine/godot/issues/35697).

Godot also have Blender exporter that allow to search for materials with the same name and apply it to the model. This is very useful but the plugin itself is experimental and lacks many features.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
It would be nice to add the ability to optionally enable search for materials with the same name in project. For example, If I have a model with Grass material, look for the material with this name in project recursively and apply if such material exists or create a new one if there is no material with the this name.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
I suggest to add this option to the Import tab as a checkbox for models. If enabled - search for materials with the same names. I also suggest to disable it by default because it can be not expected behavior.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
I often need to share materials between models but for now there is no workaround as I know. 

**Is there a reason why this should be core and not an add-on in the asset library?:**
In my opinion, such a thing should be built in. simplifies work with duplicate materials. If I need to change the Grass material, I don’t have to change it for each model separately.

**Comment by NutmegStudio** [comment]
> I suggest to add this option to the Import tab as a checkbox for models. If enabled - search for materials with the same names. I also suggest to disable it by default because it can be not expected behavior.

So if a mesh has multiple materials, there would be multiple search for materials slots?

**Comment by Shatur95** [comment]
@NutmegStudio, yes, I think that need to search for corresponding material for every slot. If I understand correctly, that is how Godot Blender Exporter is [implemented](https://docs.godotengine.org/en/3.2/getting_started/workflow/assets/escn_exporter/material.html).

**Comment by Shatur95** [comment]
I would also like to add a real example of why this feature can be very useful. I have an island landscape that is splitted into 42 chunks.And the materials used, such as Sand or Grass, are widely used in the project (I store them in a separate Materials folder). Therefore, I can’t just export the materials as separate files, because in this case they will belong only to the island. And now I need to manually apply materials for every node which is very painful:
![изображение](https://user-images.githubusercontent.com/22453358/77251835-54ded400-6c59-11ea-8e9b-515ee9045bd7.png)

**Comment by stefano-elysium** [comment]
I also encountered this as an issue, especially with model packs... 
![image](https://user-images.githubusercontent.com/85214725/162720112-505184af-7399-4564-887c-f0e56b57d6d7.png)

**Comment by Zireael07** [comment]
I can tell you I'm experiencing the same issue with asset packs in 3.x. The affected project hasn't been updated to 4.x, though, so I can't tell if the new import helps.
